### PR TITLE
Document traffic feature

### PR DESCRIPTION
**Pull request by Trietes**
# Issue

Currently there is no official documentation available about how to integrate traffic data in valhalla. See e.g. issues #3564 #3180 #2097 #1284. This PR adds a first documentation about it.

I don't have any experience with live traffic information, so I couldn't really do the documentation of it. Neither I did matching OpenLR data to valhalla. The documentation will only be about historical traffic and the mapping of OSM ids to the valhalla graph. I guess that is better then having no documentation at all.

## Tasklist

 - [x] Add #fixes with the issue number that this PR addresses
 - [x] Update the docs with any new request parameters or changes to behavior described
 - [x] Update the [changelog](CHANGELOG.md)

### CODE REVIEW

File: docs/docs/mjolnir/traffic_tiles.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# Traffic in Valhalla
+
+Traffic data for routing engines can be divided into two different types:
+
+- *Historical traffic* describes the typically observed speed on a road at a specific point in time. So e.g. the typical speed on a Monday morning at 8 o'clock.
+- *Live traffic* describes the currently observed speed on a road according to the actual traffic situation.
+
+Valhalla supports both types of traffic for all APIs, but there are some excluded use-cases. First, traffic is not supported for multimodal costing. Furthermore, traffic is currently not supported for `arrive_by` timestamps in the `source_to_targets` API. For more information about how to route with traffic data and time information, check the respective API documentation. In the following it will be explained how to integrate historical and live traffic data into the valhalla graph.
```
Comment (nilsnolde) [inline docs/docs/mjolnir/traffic_tiles.md:8]: > traffic is currently not supported for `arrive_by` timestamps in the `source_to_targets` API

That's only half true: using timedistance matrix (and less targets than sources), we do support `arrive_by`, so it's mostly a configuration item.

File: docs/docs/mjolnir/traffic_tiles.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+```
+The format of each row is `<osm_way_id>,[<direction: 0 | 1>, <graph_id>]`. Accordingly, the `osm_way_id` is mapped to multiple `graph_id`'s. An `graph_id` can be converted to the required string format according to the `to_string` function of the class `valhalla::baldr::GraphId`, whose implementation can be found in the valhalla repository. The result is a string of the following form `level/tile_id/id` (e.g. `1/47701/130`).
+
+### With OpenLR
+
+TODO
+    
+## Historical Traffic
+
+To make use of historical traffic data in valhalla, the estimated speed (in km/h) of a road segment has to be given in 5 minute intervals covering a whole week. This leads to a total of 2016 speed values per road segment. This data has to be converted to a format which can be used by valhalla and has to be structured properly in a folder hierarchy similar to the valhalla tile hierarchy. Valhalla requires the speed information in following CSV file format:
```
Comment (nilsnolde) [inline docs/docs/mjolnir/traffic_tiles.md:27]: high-res speeds is an optional field, the others however are not

File: docs/docs/mjolnir/traffic_tiles.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+```
+valhalla_tiles
+└───0
+│   │
+│   └───003
+│       │   015.gph
+│   
+└───1
+    └───047
+        │   701.gph
```
Comment (nilsnolde) [inline docs/docs/mjolnir/traffic_tiles.md:49]: you can check `tree` for linux which pretty prints directories

**Review by nilsnolde — COMMENTED** [review]
Just a few comments. Also I think we could remove the stale "Simple traffic" document here in https://github.com/valhalla/valhalla/blob/master/docs/docs/thor/simple_traffic.md

**Comment by nilsnolde** [comment]
And I'd call it "historical_traffic.md" and remove the references to live traffic/OpenLR for now.

**Comment by Trietes** [comment]
Thanks a lot for the review @nilsnolde. I just gave it some updateds according to your review. I just need to add some information at which point in time the historical speeds do start (at sunday 00:00?), so there isn't any mess with the days of the week. Then I'll give it another proof read and it would be ready from my side.

**Comment by nilsnolde** [comment]
> at which point in time the historical speeds do start (at sunday 00:00?)

Yep, the datetime library is using the stupid sunday-starts-the-week notion which even creeped into C++20: https://en.cppreference.com/w/cpp/chrono/weekday. Good point though, one of our clients had to find out that bug in our TomTom conversion pipeline where I assumed too much common sense :smile:

File: docs/mkdocs.yml [hunk 4]
```diff
@@ -105,6 +105,7 @@ nav:
       - mjolnir/map_roulette.md
       - mjolnir/map_roulette_blog.md
       - mjolnir/tag_parsing.md
+      - mjolnir/traffic_tiles.md
```
Comment (nilsnolde) [inline docs/mkdocs.yml:108]: this'll also need to change:)

**Comment by Trietes** [comment]
Its now ready from my side.

**Review by nilsnolde — APPROVED** [review]

**Comment by nilsnolde** [comment]
Oh wow, didn't think it's more or less half the world starting the week on Sunday: https://vividmaps.com/first-day-of-week-in-different-countries/. Then I'm clearly biased :laughing: I thought it's similar to Celsius/Fahrenheit, maybe a few more countries.

**Comment by Trietes** [comment]
Completely forgott about this one. Should we merge it?

**Review by RelxOff — APPROVED** [review]
thank you for adding 👍

**Comment by Trietes** [comment]
I've just updated the Changelog. It should by in-sync with the master now

**Review by nilsnolde — DISMISSED** [review]
I’ll let @kevinkreiser merge this (or anyone else), from my side this is good, thanks

**Pull request merged**